src/adapter/graphql/src/mutations/metadata_chain_mut.rs

```diff
@@ -80,6 +83,35 @@ impl MetadataChainMut {
             }
         };
 
+        use domain::DatasetRepositoryExt;
+        let dataset_repo = from_catalog::<dyn DatasetRepository>(ctx).unwrap();
+
+        let mut new_upstream_ids: Vec<opendatafabric::DatasetID> = vec![];
+
+        if let opendatafabric::MetadataEvent::SetTransform(transform) = &event {
+            for new_input in transform.inputs.iter() {
+                if let Some(id) = &new_input.id {
+                    new_upstream_ids.push(id.clone());
+                } else if let Some(dataset_ref_any) = &new_input.dataset_ref {
+                    let maybe_hdl = dataset_repo
+                        .try_resolve_dataset_ref_any(dataset_ref_any)
```

Comment:
This actually points out a big problem - we currently allow to commit `SetTransform` event with unresolved IDs. We are missing validation in MetadataChain to consider events with aliases but without IDs invalid

Created: https://github.com/kamu-data/kamu-cli/issues/366

Reply:
I think in the API we don't need the convenience of auto-resolution of aliases. Unlike when creating from YAML snapshots, users will never be writing these events by hand.

I suggest to consider ID to be always present and error out otherwise, not to attempt an alias resolution.

Reply:
Ok, simplified